packages/react-tinacms-github/src/github-client/index.ts

```diff
@@ -26,15 +36,25 @@ export class GithubClient {
   proxy: string
   baseRepoFullName: string
   baseBranch: string
-
-  constructor(
-    proxy: string,
-    baseRepoFullName: string,
-    baseBranch: string = 'master'
-  ) {
+  clientId: string
+  authCallbackRoute: string
+
+  constructor({
+    proxy,
+    clientId,
+    authCallbackRoute,
+    baseRepoFullName,
+    baseBranch = 'master',
+  }: GithubClientOptions) {
     this.proxy = proxy
     this.baseRepoFullName = baseRepoFullName
     this.baseBranch = baseBranch
+    this.clientId = clientId
+    this.authCallbackRoute = authCallbackRoute
```

Comment:
What if we check that these exist and throw an error if they don't. Then we could get rid of that new modal step. That would make it fail much earlier.

Of course the error would have to be verbose and give the same link to the docs